### PR TITLE
made this compatible with windows

### PR DESCRIPTION
**Pull request by absolute-cretin**
I was messing around with the pyls source code and I kept getting an error. Stepped through it with a debugger and realized that this line of code doesn't account for windows line endings. I fixed it so now it does work

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/python-jsonrpc-server](https://github.com/palantir/python-jsonrpc-server), @absolute-cretin! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by absolute-cretin** [comment]
Hi, I have signed the agreement :)

File: pyls_jsonrpc/streams.py [hunk 1]
```diff
@@ -25,6 +25,10 @@ def listen(self, message_consumer):
 
             if request_str is None:
                 break
+                
+            # to check for windows line endings
```
Comment (gatesn) [inline pyls_jsonrpc/streams.py:29]: I don't quite understand why this is necessary here? Doesn't the line `while line and line.strip()` strip all whitespace, including windows line endings? 

What is your rfile where you're seeing this break?

**Review by gatesn — CHANGES_REQUESTED** [review]

**Comment by delta003** [comment]
Stale

**Pull request closed without merging**